src/scalarstats.jl

```diff
@@ -336,37 +336,172 @@ end
 #   summary
 #
 #############################
+immutable SummaryStats{T<:AbstractFloat} 
+    mean::T
+    std::T
+    min::T
+    p25::T
+    median::T
+    p75::T
+    max::T
+end
 
-immutable SummaryStats{T<:AbstractFloat}
+function describe{T<:Real}(a::AbstractArray{T}, ::Type{Val{false}})
```

Comment:
@matthieugomez After having used `Val`s for a while in the linear algebra code, I've come to the conclusion that I don't like them. Do you have any ideas about alternatives? The three I can come up with are

1. Define two different functions
2. Use the detailed type for both versions, but allow uninitialized fields.
3. Ignore the type instability